std/experimental/allocator/building_blocks/free_list.d

```diff
@@ -895,42 +894,13 @@ struct SharedFreeList(ParentAllocator,
         @property void max(size_t newMaxSize);
         /// Ditto
         void setBounds(size_t newMin, size_t newMax);
-        ///
-        @safe unittest
-        {
-            import std.experimental.allocator.common : chooseAtRuntime;
-            import std.experimental.allocator.mallocator : Mallocator;
-
-            shared SharedFreeList!(Mallocator, chooseAtRuntime, chooseAtRuntime) a;
-            // Set the maxSize first so setting the minSize doesn't throw
-            a.max = 128;
-            a.min = 64;
-            a.setBounds(64, 128); // equivalent
-            assert(a.max == 128);
-            assert(a.min == 64);
-        }
```

Comment:
Unittest in templates will be part of every instantiation, they shouldn't have been there in the first place.See [DIP82](https://wiki.dlang.org/DIP82) for a more detailed description of the problem.
Also note that the unittest will still be associated with its struct.

Reply:
This can be worked around via `static if (doUnittest)` (see e.g. https://github.com/libmir/mir-algorithm/blob/e590f7f46058588f1ceb08672c92bd6248863e18/source/mir/ndslice/slice.d#L737).
Essentially you need to make sure that `doUnittest` is `true` for a single set of template arguments and that you have a template instance with those arguments (e.g. used in a private unittest).

Reply:
(If you don't want to deal with this now, we can leave it for later.)

Reply:
> This can be worked around via static if (doUnittest)

Considering that the test didn't compile until now (in case you didn't notice, the moved out version is different - it doesn't set the bounds twice as this is prohibited), I was merely trying to get things to work, s.t. we can finally move forward with the runnable examples ;-)
Regarding the `static if (doUnittest)`: I am aware of this pattern (even the old `std.experimental.ndslice` used this), but AFAIK that has been the only occurrence of this pattern in Phobos.
Anyways, there are still plenty of unittests in templated structs/classes, so I would strongly prefer if such a general cleanup is done in a later PR...

Reply:
Fair enough